Base implementation of `ModVisualAdjusts`

o-dasher opened the pull request:
I am accepting suggestions for the icon of the mod.

This pull request is just the base of this mod, it only has one setting as of now `DisableFollowPoints` i pretend to make another setting that hides the numbers of hit circles, but this most likely requires another interface to expose the circle number and some work, so i am leaving this for another pull request.

Edit:

Disable follow points discussion: https://github.com/ppy/osu/discussions/20388

Edit (2):
On top of that this pull request is now bringing the `NoComboColours` into this mod, you can see why here https://github.com/ppy/osu/pull/20514#issuecomment-1260101906

o-dasher committed:
Creates "VA" mod.

o-dasher committed:
Disable follow points setting implementation

o-dasher committed:
abstract `VisualAdjustSetting` properly

o-dasher committed:
Removes `defaultValue` option for `VisualAdjustSetting`

This because the `defaultValue` should always be false

o-dasher committed:
Creates abstract class `ModVisualAdjusts`

Since this mod may be useful for other rulesets in the future.

o-dasher committed:
Better description

mk56-spn commented:
an option to force white on combo colours maybe? some maps are heavily dependant on them ( see stuff like sputnik , where it is used to indicate rythm changes

o-dasher commented:
> an option to force white on combo colours maybe? some maps are heavily dependant on them ( see stuff like sputnik , where it is used to indicate rythm changes ) and quite a few ez players use them iirc for pattern recognition

I don't get what you mean but i would rather keep it simple just to keep things more scoped and easier to review but this could be implemented on another pull request though, do you have an video example of waht you mean with white combo colours?

LeNitrous commented:
Shouldn't hit object elements be controlled at a skinning level instead of being a mod? Even if mod presets exist, it can get messy having to toggle the mod or the preset containing the mod every time.

o-dasher commented:
> Shouldn't hit object elements be controlled at a skinning level instead of being a mod? Even if mod presets exist, it can get messy having to toggle the mod or the preset containing the mod every time.

Read the discussion, this is exclusive to things that may affect difficulty.

mk56-spn commented:
> i think a screenshot shall suffice, the colours for new combos, give players a certain info, making the circles easier to group especially on low ars
![osu_2022-09-27_23-59-35](https://user-images.githubusercontent.com/74463310/192644700-a18425a1-ccdc-4f74-a9b7-fe89ebaee6f7.png)
heres what i mean . now with all new combo colours switched to white a lot of information is  no longer available to the player. 
![osu_2022-09-28_00-03-45](https://user-images.githubusercontent.com/74463310/192645891-f79b8e4a-0ac8-4b20-aab2-bd2dff790dc0.png)
but yes i agree it can be done later

o-dasher commented:
Oh noted.

bdach commented on osu.Game/Rulesets/Mods/ModVisualAdjusts.cs:
```diff
@@ -0,0 +1,53 @@
+        public override string Acronym => "VA";
+        public override ModType Type => ModType.Conversion;
+
+        private void triggerAdjustsForType<B, A>(A args) where B : VisualAdjustSetting<A>
```
This looks like pretty extreme overengineering to me. Doubly-bound generics? For what reason? There's only one usage ever, can we start simpler and see where this goes rather than lock into specific structures this early?

Also if these generic args *are* to stay, please name them better. In C# land we generally don't do the rust-like single letter generic alias thing. Convention is usually `TTypeName` (as in, a longer, *descriptive* name for the alias, preceded by a `T`).

o-dasher replied:
I would rather keep this structure with the generics in place because i already know a use case that i will soon implement (Hide the combo numbers as mentioned) this structure allows for this implementation be way easier.
Would you like me to add another usage? like remove combo colors since that is most likely easier to implement.

bdach replied:
The only way I can be personally convinced is if you show me at least one more usage using this exact pattern, and _also_ if you show that a naive, simple implementation using no generics is clearly inferior/much longer. If you're doing this to save 5 extra lines of code, then I argue that it's not worth the complexity overhead.

o-dasher replied:
Check latest commit, added an use case for "WhiteComboColours" setting.
But here is the comparison for generic vs non-generic.

Generic:
```cs
        private void triggerAdjustsForType<TBindable, TArgs>(TArgs args) where TBindable : VisualAdjustSetting<TArgs>
        {
            foreach (var (_, property) in this.GetOrderedSettingsSourceProperties())
            {
                if (property.GetValue(this) is TBindable bindable && bindable.Value)
                    bindable.ApplyAdjusts(args);
            }
        }

        public void ApplyToDrawableRuleset(DrawableRuleset<TObject> drawableRuleset)
        {
            triggerAdjustsForType<DrawableRulesetVisualAdjustSetting, TDrawableRuleset>((TDrawableRuleset)drawableRuleset);
        }

        public void ApplyToDrawableHitObject(DrawableHitObject drawable)
        {
            triggerAdjustsForType<DrawableHitObjectVisualAdjustSetting, DrawableHitObject<TObject>>((DrawableHitObject<TObject>)drawable);
        }
```
Non-generic:
```cs
        public void ApplyToDrawableRuleset(DrawableRuleset<TObject> drawableRuleset)
        {
            var tDrawableRuleset = (TDrawableRuleset)drawableRuleset;

            foreach (var (_, property) in this.GetOrderedSettingsSourceProperties())
            {
                if (property.GetValue(this) is DrawableRulesetVisualAdjustSetting bindable && bindable.Value)
                    bindable.ApplyAdjusts(tDrawableRuleset);
            }
        }

        public void ApplyToDrawableHitObject(DrawableHitObject drawable)
        {
            var tDrawable = (DrawableHitObject<TObject>)drawable;

            foreach (var (_, property) in this.GetOrderedSettingsSourceProperties())
            {
                if (property.GetValue(this) is DrawableHitObjectVisualAdjustSetting bindable && bindable.Value)
                    bindable.ApplyAdjusts(tDrawable);
            }
        }
```
And add another method for every new use case... the generic for me sure looks way better now.

o-dasher replied:
I was able to simplify generics usage on latest commit must be even clearer now.

o-dasher committed:
Proper name generic types to fit c# land styling

o-dasher committed:
Adds `DrawableHitObjectVisualAdjustSetting` i

o-dasher committed:
Adds `WhiteComboColours` setting.

mk56-spn commented on osu.Game.Rulesets.Osu/Mods/OsuModVisualAdjusts.cs:
```diff
@@ -0,0 +1,20 @@
+        [SettingSource("Disable follow points", "No more hints for where to follow...")]
+        public DrawableRulesetVisualAdjustSetting DisableFollowPoints { get; } = new DrawableRulesetVisualAdjustSetting(ruleset => ruleset.Playfield.FollowPoints.Hide());
+
+        [SettingSource("White combo colours", "All combos will now be white!")]
```
I'd probably call it "no combo colours" Or something of the sort. Since white is just white as a sort of "non colour". If you get what I mean

o-dasher replied:
Yes, you are right.

o-dasher replied:
l

done.

o-dasher committed:
Rename `WhiteComboColours` to `NoComboColours`

o-dasher committed:
Simplify generics usage.

o-dasher commented:
The no colours feature seems to not be working as expected, i probably tested on a map which had white combo, i will take a look at it.

o-dasher committed:
Removes unused generic cast

peppy commented on osu.Game.Rulesets.Osu/Mods/OsuModVisualAdjusts.cs:
```diff
@@ -0,0 +1,20 @@
+{
+    internal class OsuModVisualAdjusts : ModVisualAdjusts<OsuHitObject, DrawableOsuRuleset>
+    {
+        [SettingSource("Disable follow points", "No more hints for where to follow...")]
+        public DrawableRulesetVisualAdjustSetting DisableFollowPoints { get; } = new DrawableRulesetVisualAdjustSetting(ruleset => ruleset.Playfield.FollowPoints.Hide());
+
+        [SettingSource("No combo colours", "The combo colours won't tell you anything now...")]
+        public DrawableHitObjectVisualAdjustSetting NoComboColours { get; } = new DrawableHitObjectVisualAdjustSetting(dho => dho.AccentColour.Value = Colour4.White);
+    }
```
I'm *so confused by this*. There's too much magic. Please write this similar to other mods and do away with the magic.

o-dasher replied:
I mean there isn't much magic here, a `VisualAdjustSetting` is just a setting that handles what to do when the setting is active, wouldn't documenting be enough?

o-dasher replied:
Just noticed that a simple structure will do better... both optimization wise and for the developer i guess, will make it when i get home

o-dasher replied:
I take back on this i will try implementing what i think should be the most understandable when i get home.

o-dasher replied:
> I'm *so confused by this*. There's too much magic. Please write this similar to other mods and do away with the magic.

Could you take a look at my latest commit, although the structure follow the same pattern it is better explained now i believe, and it is now handling the `NoComboColours` part correctly.

o-dasher committed:
Properly setup `ModWithVisibilityAdjustment` for `ModVisualAdjust`

o-dasher committed:
Adds simple documentation

o-dasher committed:
Simplify visual settings setup flow

o-dasher commented:
First of all i am sorry i am still getting used to the code base, i was doing all that over engineering on the though that some used methods would be called multiple times on gameplay since this isn't the case at all i am making things much simpler now :(

o-dasher committed:
Completely removes overengineered jank

mk56-spn commented:
> White combo color is a bit hard to read on classic default skin, maybe pick the first combo color in the skin?

Sounds like a good idea. But that could also be white. Hmm

o-dasher commented:
> White combo color is a bit hard to read on classic default skin, maybe pick the first combo color in the skin?

Not sure on this it would make the setting being based on your skin which makes it weird from a user perspective, i would rather let it be white (white as no color) and if you can't read with your skin just switch it as trivial as that

bdach committed:
Add other profile sections to test scene

bdach committed:
Adjust general appearance of user profile overlay

bdach committed:
Adjust appearance of section navigation

bdach committed:
Adjust appearance of user profile sections

bdach committed:
Rearrange and adjust header components where simple

bdach committed:
Adjust visual appearance of profile value displays

bdach committed:
Adjust user profile header detail appearance

bdach committed:
Adjust colour of top header container

bdach committed:
Move contents of detail header to separate component

bdach committed:
Move out stats from top header container

bdach committed:
Move user stats into detail header container

bdach committed:
Update colouring of some profile section elements

bdach committed:
Fix wrong date colour of kudosu history items

mk56-spn commented on osu.Game.Rulesets.Osu/Mods/OsuModVisualAdjusts.cs:
```diff
@@ -0,0 +1,45 @@
+
+        protected override void ApplyNormalVisibilityState(DrawableHitObject hitObject, ArmedState state)
+        {
+            tryApplyNoComboColours(hitObject);
```
Also I'd probably use arrow syntax here considering it's a single statement and clarity wouldn't be affected.

o-dasher replied:
These shouldn't even be there in the first place since it was doing the same for all "visibility states", Fixed :P

mk56-spn commented:
Just a couple of nitpicks I have upon checking this out again. Nothing major

o-dasher committed:
Remove unnecessary methods.

peppy commented:
I think this should be called something that better describes that it is obfuscating visual cues. It's basically always increasing the difficulty, so the name should reflect that.

mk56-spn commented:
> I think this should be called something that better describes that it is obfuscating visual cues. It's basically always increasing the difficulty, so the name should reflect that.

"impeded" perhaps?. i feel it gets the point across quite well that you are in some way handicapping yourself through the removal of visual cues as you said. perhaps a little on the nose. all the other examples i can come up with feel they are just generally removing visibility, whilst this seemed to fit

o-dasher commented:
Yeah, impeded seems to work fine

bdach commented:
Left field name suggestion: what about "Incognito"?

![1673281101](https://user-images.githubusercontent.com/20418176/211355741-623f2b50-e821-4437-85d1-04b3544a60ee.png)

Kinda fits the theme of hiding combo colours. Follow point less so, but I guess it also obscures things visually somewhat, so maybe it kinda works?

mk56-spn commented:
> Left field name suggestion: what about "Incognito"?
> 
> ![1673281101](https://user-images.githubusercontent.com/20418176/211355741-623f2b50-e821-4437-85d1-04b3544a60ee.png)
> 
> Kinda fits the theme of hiding combo colours. Follow point less so, but I guess it also obscures things visually somewhat, so maybe it kinda works?

on the one hand it does sound amazing and vaguely fits, but IMO incognito would be better fitted to a hypothetical implementation of a " fake notes " mod people have wanted for quite a while, that said im not totally opposed, after all  i think form is more important than function in mod naming, since if you ask me the important thing in making mods stick is the name being unique, not so much it being spot on  dictionary accurate

bdach commented:
> IMO incognito would be better fitted to a hypothetical implementation of a " fake notes " mod people have wanted for quite a while

I can think of a few other names for something like that. "Carbon Copy", "Impostor" (though that one is maybe a bit meme-tainted now...), "Double Vision"...

peppy committed:
Fix kiai flash opacity on legacy skins being too intense

peppy committed:
Add full colour application to kiai sprites

peppy committed:
Fix previous badges potentially not being cleared if new user has no badges

peppy committed:
Remove some redundancies

peppy committed:
Merge pull request #22122 from Stedoss/beatmapsetoverlay-fetch-on-login

Fix `BeatmapSetOverlay` not showing a pending request after logging in

peppy committed:
Fix skin fail sound not correctly playing

Closes #21719. Tested using skin in issue thread.

peppy committed:
Add extra level of nullabiliy checking because NRT is not present

peppy committed:
Fix incorrect application layer causing completely discoloured circles

peppy committed:
Merge pull request #22123 from stanriders/group-badges

Implement user group badges

ItsShamed committed:
perf: Do not draw sectoins that are less than 1px

ItsShamed committed:
refactor: make class not abstract

ItsShamed committed:
fix(SegmentedGraph): make ctor public

ItsShamed committed:
fix(SegmentedGraph): solve issue for negatives values

ItsShamed committed:
test: add visual tests for SegmentedGraph class

ItsShamed committed:
test(SegmentedGraph): add concrete application test with beatmap density

bdach committed:
Merge branch 'master' into fix-fail-sound-not-playing

bdach committed:
Merge pull request #22133 from peppy/fix-fail-sound-not-playing

Fix skin fail sound not correctly playing

bdach committed:
Merge branch 'master' into user-profile/ruleset-switching

Feodor0090 committed:
Add failing test

frenzibyte committed:
Fix game-level components finishing load after content

ItsShamed committed:
style: nitpicks

bdach committed:
Rewrite contributing guidelines

- Includes changes to process discussed in development meeting today.
- Describes issue vs discussion axis which wasn't present at the time
  the document was originally created.
- Reduces amount of unnecessary text in hopes that the document will
  actually be read in full.

Feodor0090 committed:
More tests

Feodor0090 committed:
Make snapped time always positive

Feodor0090 committed:
Fix unrelated tests

ItsShamed committed:
fix(SegmentedGraph): update `graphNeedsUpdate` variable during `Update()` loop

Joehuu committed:
Add failing beatmap set lookup type test

Joehuu committed:
Fix lookup type being incorrect when fetching beatmap set

peppy committed:
Update README slightly to work better with new document

smoogipoo committed:
Merge pull request #22136 from peppy/fix-kiai-flash-intensity

Fix kiai flash opacity on legacy skins being too intense

smoogipoo committed:
Merge pull request #22149 from Joehuu/fix-beatmap-set-lookup-type

Fix lookup type being incorrect when fetching beatmap set

peppy committed:
Merge pull request #22146 from bdach/rewrite-contributing-guidelines

Rewrite contributing guidelines

peppy committed:
Merge pull request #22006 from bdach/user-profile/ruleset-switching

Add ruleset switching to user profile overlay

smoogipoo committed:
Fix incorrect max combo after watching imported legacy replays

frenzibyte committed:
Avoid changing target content midway through code

peppy committed:
Merge pull request #22141 from frenzibyte/fix-game-load-order

Fix game-level components not finishing load before game content

ItsShamed committed:
refactor(SegmentedGraph): use (get/set)ters to expose TierColour

peppy committed:
Merge pull request #22151 from smoogipoo/fix-incorrect-legacy-combo

Fix incorrect max combo after watching imported legacy replays

ItsShamed committed:
refactor(SegmentedGraph): adjust tierCount based on passed Colours

peppy committed:
Merge branch 'master' into ui/segmented-graph

peppy committed:
Merge pull request #22143 from ItsShamed/ui/segmented-graph

Implement a segmented graph

frenzibyte committed:
Mark item-mutating queue mode tests with `FlakyTest` attribute

peppy committed:
Remove `CFBundleVersion` from iOS plist file

This allows using `-p:ApplicationVersion:1234.5.6` to specify a version
at build / publish time. Without removing it the plist file's value
always takes precedence.

peppy committed:
Also remove `CFBundleShortVersionString`

Feodor0090 committed:
Enable nullability for editor components

Feodor0090 committed:
Undo sizing changes

EVAST9919 committed:
Make OverlayRulesetSelector display only legacy rulesets

EVAST9919 committed:
Don't display non-legacy rulesets in beatmap listing

frenzibyte committed:
Fix channel manager test scene not waiting for notifications client connection

EVAST9919 committed:
Use IsLegacyRuleset extension method

EVAST9919 committed:
Use property instead of ctor parameter

Joehuu committed:
Fix mute button not blocking outside overlay closing presses

Feodor0090 committed:
Add "fps" keyword to frame limiter dropdown

mk56-spn committed:
Enable nullability for song select

peppy committed:
Merge pull request #22170 from Feodor0090/fps-keyword

Add "fps" keyword to frame limiter dropdown

peppy committed:
Merge pull request #22169 from Feodor0090/timeline-nullability

Enable nullability for editor components

peppy committed:
Merge pull request #22163 from frenzibyte/fix-intermittent-channel-test-failure

Fix intermittent failure in silenced user messages test

peppy committed:
Fix grammar

peppy committed:
Merge pull request #22160 from EVAST9919/ruleset-selector-online

Don't display custom rulesets in online overlays

peppy committed:
Add default version specifications to iOS csproj file

Matches implementation for android.

Joehuu committed:
Add comment highlighting reason for blocking mouse down

peppy committed:
Merge pull request #22168 from Joehuu/fix-mute-button-closing-overlays

Fix mute button not blocking outside overlay closing presses

peppy committed:
Merge branch 'master' into remove-cf-bundle-version

peppy committed:
Enable NRT on account creation classes

peppy committed:
Use more correct default version to avoid startup crash on attempting to parse

peppy committed:
Add API level support for error message and redirect during registration flow

peppy committed:
Add UI support for redirecting the user to web registration

peppy committed:
Prefill username and email

peppy committed:
Fix debug build thinking it's deployed due to major version >= 1

smoogipoo committed:
Merge pull request #22172 from peppy/create-account-redirect-support

Add support for redirection to website during account creation process

mk56-spn committed:
Fix test failures

smoogipoo committed:
Merge pull request #22156 from peppy/remove-cf-bundle-version

Remove `CFBundleVersion` from iOS plist file

mk56-spn committed:
Make carousel non nullable and ensure pre load usages of methods that reference it are protected against failure

Mogiiii committed:
Use PlacementState to check juice stream placement

Mogiiii committed:
Merge branch 'master' of https://github.com/ppy/osu into 18469

Mogiiii committed:
Code Quality

frenzibyte committed:
Fix intermittent failure in certain beatmap carousel tests

frenzibyte committed:
Add leading zeros to test beatmaps for correct title sorting

frenzibyte committed:
Fix potential nullref in certain beatmap carousel tests

frenzibyte committed:
Fix intermittent failure in score submission tests

Feodor0090 committed:
Add localisation for notification overlay

Feodor0090 committed:
Localise some notifications

Feodor0090 committed:
Remove `CanBeNull` where it's no longer requered

stanriders committed:
Add `StatusIcon`

Joehuu committed:
Move overlay ruleset selectors to tab control

bdach committed:
Merge pull request #22186 from frenzibyte/fix-score-test-failure

Fix intermittent failure in score submission tests

Feodor0090 committed:
Use existing string

Feodor0090 committed:
Autoformat code

bdach committed:
Merge branch 'master' into comment-editor-1

stanriders committed:
Set border colour to white

bdach committed:
Merge pull request #21448 from Feodor0090/comment-editor-1

Update comment editor component

integerrr committed:
revert additions to `SessionStatics`

integerrr committed:
make `PlayerSettingsGroup` expand on hover

stanriders committed:
Fix `GroupBadge` crashing on `null` group colour

Feodor0090 committed:
Enable nullability for timing screen

Feodor0090 committed:
Enable nullability for setup screen

Feodor0090 committed:
Enable nullability for design screen

stanriders committed:
Using

Feodor0090 committed:
Adjust naming

stanriders committed:
Use `StatusIcon` in `UserDropdown`

Feodor0090 committed:
Merge branch 'master' into comment-editor-3

Feodor0090 committed:
Adapt to changes in base class

Feodor0090 committed:
Revert "Use existing string"

This reverts commit f950b624ae4f0659f12f43148b2782aff60b98df.

Feodor0090 committed:
Move avatar update bind to LoadComplete

stanriders committed:
Using

Joehuu committed:
Add ability to view converted beatmaps on beatmap set overlay

bdach committed:
Merge pull request #22192 from stanriders/fix-group-badges

Fix `GroupBadge` crashing on `null` group colour

peppy committed:
Merge pull request #22194 from Feodor0090/more-editor-nrt

Enable nullability for setup, design and timing editor screens

peppy committed:
Merge pull request #22188 from Feodor0090/nofs-locale

Localise some notifications

ekrctb committed:
Merge branch 'master' into 18469

ekrctb committed:
Merge pull request #21522 from Mogiiii/18469

Show distance snap for first juice stream object.

frenzibyte committed:
Fix intermittent failure in GCC seeking test

peppy committed:
Merge pull request #22182 from stanriders/status-icon

Update status icon design

bdach committed:
Merge pull request #22198 from frenzibyte/fix-mgcc-test-failure

Fix intermittent failure in GCC seeking test

integerrr committed:
fix wrong control flow on hover lost

bdach committed:
Merge branch 'master' into move-overlay-ruleset-selectors

bdach committed:
Merge pull request #22191 from Joehuu/move-overlay-ruleset-selectors

Move overlay ruleset selectors to tab control in line with web

Feodor0090 committed:
Rename parent comment id field

peppy committed:
Merge pull request #22181 from frenzibyte/attempt-fix-song-select-test-failure

Fix intermittent failure in beatmap carousel tests

bdach committed:
Add failing test for new difficulty creation

peppy committed:
Merge pull request #22200 from bdach/user-profile/update-design-basics

Update user profile appearance to closer match web

Feodor0090 committed:
Add test case with no existing comments

bdach committed:
Fix difficulty creation flow failing for some ruleset combinations

In current `master`, the difficulty creation flow would retrieve a
"reference beatmap" to copy metadata and timing points from using
`GetPlayableBeatmap()`. But, importantly, it was calling that method
using *the ruleset of the new difficulty* rather than the ruleset of the
existing one. This would make the difficulty creation flow fail if the
beatmap couldn't be converted between rulesets (like taiko to catch).

Fixing to use the reference beatmap's actual ruleset would be trivial,
but there's no real reason to do all of that work anyway when a
`WorkingBeatmap` is available. Because getting a playable beatmap is not
required to copy across metadata and timing points, remove the
`GetPlayableBeatmap()` step entirely to fix the bug.

Closes #22145.

Feodor0090 committed:
Add check for any drawables in content before looking for placeholder

Feodor0090 committed:
Take type and id from container directly on submit

Feodor0090 committed:
Use clock to obtain a position for comment insertion

Feodor0090 committed:
Make test wait for comment loading

Feodor0090 committed:
Explain empty string in editor's footer

integerrr committed:
make expansion to not use scheduler

Feodor0090 committed:
Add support for localisation where it was missing

Feodor0090 committed:
Localise background header

Feodor0090 committed:
Localise create/save dialog

Feodor0090 committed:
Localise part of editor

Joehuu committed:
Fix kudosu wiki link from user profile not linking to in-game overlay

peppy committed:
Merge pull request #22213 from Joehuu/fix-kudosu-wiki-link

Fix kudosu wiki link from user profile not linking to in-game overlay

ekrctb committed:
Remove #nullable disable from Catch.Edit namespace

ekrctb committed:
Remove #nullable disable from Catch.Beatmaps

ekrctb committed:
Remove #nullable disable from Catch.Difficulty

ekrctb committed:
Remove #nullable disable from Catch.Judgements

ekrctb committed:
Remove #nullable disable from Catch.Objects

ekrctb committed:
Remove #nullable disable from misc Catch files

ekrctb committed:
Remove #nullable disable from Catch.UI

ekrctb committed:
Remove #nullable disable from Catch.Objects.Drawables

Except DrawableCatchHitObject, it complains in base(hitObject) call.

Feodor0090 committed:
Use existing strings

Feodor0090 committed:
Remove text padding in editor menu for now

frenzibyte committed:
Fix intermittent failure in tests with restarting player instances

peppy committed:
Revert some formatting changes

peppy committed:
Make `MusicController` a required dependency of `SongSelect`

peppy committed:
Merge branch 'master' into song_select_nullability

ekrctb committed:
Remove redundant `canBeNull: true`

ekrctb committed:
Remove #nullable disable from CaughtObject

mk56-spn committed:
remove hacky method to check if carousel is null

bdach committed:
Merge pull request #22218 from ekrctb/nullable-catch

Remove #nullable disable from osu.Game.Rulesets.Catch

Feodor0090 committed:
Move some strings to common

peppy committed:
Merge pull request #22221 from frenzibyte/fix-screen-navigation-test-failure

Fix intermittent failure in screen navigation tests

Feodor0090 committed:
Localise osu! settings

Feodor0090 committed:
Localise osu!mania settings

Joehuu committed:
Mark `Converts` as nullable

o-dasher commented:
I think i personally prefer the incognito route, if not any new suggestions i will change it and make the pr up-to date

Joehuu committed:
Apply NRT to `BeatmapPicker`

Joehuu committed:
Move converted beatmap icons logic locally

Joehuu committed:
Merge remote-tracking branch 'upstream/master' into pr/20709

Joehuu committed:
Add argument for play some ruleset string

Joehuu committed:
Make home string keys more specific

peppy committed:
Rename ambiguous translation keys

peppy committed:
Use scroll speed localisation in one more usage

peppy committed:
Merge branch 'master' into ruleset-settings-loc

peppy committed:
Merge pull request #22225 from Feodor0090/ruleset-settings-loc

Localise ruleset settings section

frenzibyte committed:
Fix intermittent failure in beatmap options state test

smoogipoo committed:
Merge pull request #22232 from frenzibyte/fix-beatmap-options-test-failure

Fix intermittent failure in beatmap options state test

peppy committed:
Merge pull request #22208 from Feodor0090/editor-loc

Localise most visible parts of editor

Feodor0090 committed:
Localise "revert to default" tooltip

Feodor0090 committed:
Localise menu bar

Feodor0090 committed:
Localise editor's UI

frenzibyte committed:
Mark ready-button-pressing multiplayer tests with `FlakyTest` attribute

bdach committed:
Merge branch 'master' into beatmap-set-converted-beatmaps

Feodor0090 committed:
Select button using its index in test

Feodor0090 committed:
Do not touch "currently editing" line

stanriders committed:
Refactor `LevelBadge` to use `LevelInfo`

bdach committed:
Merge pull request #22206 from Joehuu/beatmap-set-converted-beatmaps

Add ability to view converted beatmaps on beatmap set overlay

stanriders committed:
Rename variable

Co-authored-by: Bartłomiej Dach <dach.bartlomiej@gmail.com>

stanriders committed:
Move binding to `LoadComplete`

bdach committed:
Move to `LoadComplete()` better

bdach committed:
Merge pull request #22241 from stanriders/refactor-levelbadge

Refactor `LevelBadge` to use `LevelInfo`

peppy committed:
Use `ToString()` for test instead of linq skip

peppy committed:
Merge branch 'master' into skin-editor-loc

peppy committed:
Merge pull request #22236 from Feodor0090/skin-editor-loc

Localise skin editor

peppy committed:
Merge pull request #22203 from bdach/fix-new-diff-creation-fail

Fix difficulty creation flow failing for some ruleset combinations

peppy committed:
Merge branch 'master' into replay-menu-expanded-state-memory

peppy committed:
Move hover logic to `SettingsToolboxGroup` to avoid expanded state clash

peppy committed:
Merge pull request #21829 from integerrr/replay-menu-expanded-state-memory

Make replay controls menus expand on hover

peppy committed:
Avoid need for weird null check by reordering initialisation process

peppy committed:
Merge branch 'master' into song_select_nullability

peppy committed:
Merge pull request #22246 from frenzibyte/mark-tests-flaky

Mark remaining frequently-failing intermittent tests with `FlakyTest` attribute

Joehuu committed:
Simplify container logic

Joehuu committed:
Use `TriggerClick()` instead of calling `BackButton` action for pause gameplay

peppy commented:
Incognito sounds good, I think.

peppy committed:
Merge pull request #22174 from mk56-spn/song_select_nullability

Enable NRT annotations in `SongSelect` classes

peppy committed:
Merge pull request #20709 from Domilz/toolbar-localisation

Localise toolbar home button and ruleset description tooltips

Joehuu committed:
Simplify `TriggerClick`s in gameplay menu overlays

peppy committed:
Merge pull request #22147 from Feodor0090/forbid-negative-snap

Forbid negative time snap in editor

peppy committed:
Merge branch 'master' into fix-pause-gameplay-action-not-closing

peppy committed:
Merge pull request #22079 from Joehuu/fix-pause-gameplay-action-not-closing

Make pause overlay close with pause gameplay action

peppy committed:
Always use production (non-experimental) endpoints for tournament client

peppy committed:
Fix tournament client not saving changes after populating new data

peppy committed:
Fix player rank data not being re-fetched after a tournament's ruleset is changed

ekrctb committed:
Reduce number of tested value combination
121 combinations took 2min+

bdach committed:
Fix code quality inspection

bdach committed:
Merge pull request #22256 from peppy/tournament-repopulate-ranks-after-ruleset-change

Fix player rank data not being re-fetched after a tournament's ruleset is changed

bdach committed:
Merge branch 'master' into tournament-fix-save-changes-on-population

bdach committed:
Merge pull request #22255 from peppy/tournament-fix-save-changes-on-population

Fix tournament client not saving changes after populating new data

bdach committed:
Merge branch 'master' into tournament-always-use-production-endpoint

bdach committed:
Merge pull request #22254 from peppy/tournament-always-use-production-endpoint

Always use production (non-experimental) endpoints for tournament client

bdach committed:
Add logging on comment post failure

bdach committed:
Merge branch 'master' into comment-editor-3

smoogipoo committed:
Merge pull request #22258 from ekrctb/remove-slow-test

Reduce number of value combinations of a test that is taking 2min+ to execute

bdach committed:
Merge pull request #22193 from Feodor0090/comment-editor-3

Integrate comment editor into comments view

o-dasher committed:
Rename to ModIncognito

o-dasher committed:
Rename to ModIncognito

smoogipoo committed:
Merge pull request #22266 from bdach/user-profile/update-design-stats

Update appearance of user statistics on profile overlay to match web

mk56-spn commented on osu.Game.Rulesets.Osu/Mods/OsuModIncognito.cs:
```diff
@@ -12,7 +12,7 @@
-    internal class OsuModVisualAdjusts : ModVisualAdjusts, IApplicableToDrawableRuleset<OsuHitObject>, IApplicableToDrawableHitObject
+    internal class OsuModIncognito : ModIncognito, IApplicableToDrawableRuleset<OsuHitObject>, IApplicableToDrawableHitObject
     {
         [SettingSource("Disable follow points", "No more hints for where to follow...")]
```
Not a fan of this description it feels somewhat clumsily formulated. 

I'd opt for either something tongue in cheek... say... " The lines... where are they?" or " disconnected"

Given the main setting name is already very clear, so having some fun with the description seems appropriate. 

Even if that isn't the desired path id still rephrase it as something like " No hints on where to go next.." or " Find your way from circle to circle unaided"

o-dasher replied:
Yeah, i was waiting for someone to point out this it seemed odd to me too, is just that i couldn't think of a way to rephrase it.
I will go with "The lines... where are they?'

mk56-spn commented on osu.Game/Rulesets/Mods/ModIncognito.cs:
```diff
@@ -0,0 +1,18 @@
+    public abstract class ModIncognito : Mod
+    {
+        public override string Name => "Incognito";
+        public override LocalisableString Description => "Obscure some gameplay visual elements.";
```
"Visual" feels redundant here and makes the sentence somewhat irksome in my humble opinion, at least in the way its currently phrased. This is doubly true since the word "obscure" as a verb already invokes a meaning akin to " limited vision " of a given thing. 

"obscure the visuals" is another workable alternative if you want to double down on the visual side, since the fact that its alluding to the gameplay is plenty clear from context alone

mk56-spn reviewed (commented):
Nitpicks

o-dasher committed:
Apply suggestions.

o-dasher committed:
Merge remote-tracking branch 'origin/visual-adjusts-mod' into visual-adjusts-mod

# Conflicts:
#	osu.Game.Rulesets.Osu/Mods/OsuModIncognito.cs
#	osu.Game/Rulesets/Mods/ModIncognito.cs

o-dasher commented:
Sorry. I am not sure what i did but this isn't right...
Maybe it was the rebase when making the branch up to date
